File: staylio_backend/src/main/java/com/example/staylio_backend/dto/response/RoomSearchResponse.java
package com.example.staylio_backend.dto.response;

import com.example.staylio_backend.model.enums.RoomStatus;
import com.example.staylio_backend.model.enums.RoomType;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@Builder
public class RoomSearchResponse {
    private Long roomId;

    private String roomName;

    private RoomType roomType;

    private List<String> images;

    private Long hotelId;

    private String hotelName;

    private Long hotelBranchId;

    private String hotelBranchName;

    private String address;

    private String provinceName;

    private Integer capacity;

    private Integer maxAdults;

    private Integer maxChildren;

    private BigDecimal price;

    private Double averageRating;

    private Long reviewCount;

    private RoomStatus status;
}

File: staylio_frontend/common/interfaces/response/RoomSearchResponse.ts
import { RoomStatus } from "@common/enums/RoomStatus";
import { RoomType } from "@common/enums/RoomType";

export interface RoomSearchResponse {
    roomId: number;
    roomName: string;
    roomType: RoomType;
    images: string[];
    hotelId: number;
    hotelName: string;
    hotelBranchId: number;
    hotelBranchName: string;
    address: string;
    provinceName: string;
    capacity: number;
    maxAdults: number;
    maxChildren: number;
    price: number;
    averageRating: number;
    countReview: number;
    status: RoomStatus
}